wordpress-mobile/WordPress-Android
# Add "create a site" deeplink handling

Fixes #14371 

This solution introduces handling for the `public-api.wordpress.com/mbar/...redirect_to=wordpress.com/wp-login.php?...redirect_to="wordpress.com/start"` links from emails and opens a site creation flow when these links are clicked.
There is now a view model which handles all the `mbar` links and all the links that open the editor (just to have them in one place). 

To test - Site creation:
- Log in in the app with a WP.com account
- Click on the "Create your site" in an email (let me know if you need the link from the email)
- Notice this line of code in the LogCat - "DeepLink tracking URI successfully requested"
- Notice the "Site creation" flow is started
- Finish the flow and create a site
- Notice you end up on the My Site screen with the site is selected 

To test 2 - Site creation not logged in
- Do not log in to the app with a WP.com account
- Click on the "Create your site" in an email (let me know if you need the link from the email)
- Notice this line of code in the LogCat - "DeepLink tracking URI successfully requested"
- Notice the "WP.com signup" flow is started
- Log in/signup
- You end up logged in on the My Site screen

To test 3 - edit an existing post
- Log in to the app
- Click on a link like `https://wordpress.com/post/site_url/post_id` that exists for a site on an account you're logged in with 
- Notice this line of code in the LogCat - "DeepLink tracking URI successfully requested"
- Notice the post is opened in the editor

To test 4 - edit a non-existing post
- Log in to the app
- Click on a link like `https://wordpress.com/post/site_url/post_id` that doesn't exists for a site on an account you're logged in with 
- Notice this line of code in the LogCat - "DeepLink tracking URI successfully requested"
- Notice the editor is opened for site with site_url

To test 5 - edit a non-existing post on a non-logged in site
- Log in to the app
- Click on a link like `https://wordpress.com/post/site_url/post_id` that doesn't exists for a site that you're not logged in
- Notice this line of code in the LogCat - "DeepLink tracking URI successfully requested"
- Notice the editor is opened for your current site

## Regression Notes
1. Potential unintended areas of impact
- All the `mbar` links are impacted but they should all be handled as before

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Unit tested the classes that could be tested

3. What automated tests I added (or what prevented me from doing so)
- unit tests in this PR

PR submission checklist:
